packages/codemods/src/legacy-compat-builders/legacy-store-method.ts

```diff
@@ -0,0 +1,201 @@
+          message: error.message,
+        });
+        return;
+      }
+      throw error;
+    }
+
+    // Replace with, e.g. store.request(findRecord('post', '1'))
+    // First, change the callee to store.request
+    path.value.callee.property.name = 'request';
+
+    // Then, wrap the arguments with the builder expression
+    const builderExpression = j.callExpression.from({
+      callee: j.identifier(importInfo.localName ?? importInfo.importedName),
+      arguments: path.value.arguments,
+    });
+
+    const builderTypeParameters = extractBuilderTypeParams(j, path);
+    path.value.typeParameters = null;
+    // SAFETY: JSCodeshift types are wrong
+    (builderExpression as unknown as { typeParameters: TSTypeParameterInstantiation | null }).typeParameters =
+      builderTypeParameters;
+    path.value.arguments = [builderExpression];
+
+    if (isRecord(path.parent.parent)) {
+      if (
+        j.VariableDeclarator.check(path.parent.parent.value) &&
+        (j.Identifier.check(path.parent.parent.value.id) || j.ObjectPattern.check(path.parent.parent.value.id))
+      ) {
+        // Replace `const post` with `const { content: post }`
+        // Replace `const { id }` with `const { content: { id } }`
+        const id = path.parent.parent.value.id;
+        // Intentionally drop unnecessary type annotation from id, as it causes syntax errors otherwise
+        const value = j.Identifier.check(id)
+          ? j.identifier.from({ ...id, typeAnnotation: null })
+          : j.objectPattern.from({ ...id, typeAnnotation: null });
+        path.parent.parent.value.id = j.objectPattern.from({
+          properties: [
+            j.objectProperty.from({
+              key: j.identifier.from({ name: 'content' }),
+              value,
+            }),
+          ],
+        });
+      } else if (j.ReturnStatement.check(path.parent.parent.value)) {
+        // It's not assigned to a variable so we don't need to worry about destructuring
+        // but we do need to make sure the value stays the same, so:
+        // Wrap the whole await expression in a MemberExpression to add `.content`
+        const memberExpression = j.memberExpression.from({
+          object: path.parent.value,
+          property: j.identifier.from({ name: 'content' }),
+        });
+
+        // Finally, replace
+        j(path.parent).replaceWith(memberExpression);
+      }
+    }
+
+    result.importsToAdd.add(importInfo);
+  });
+
+  return result;
+}
+
+function legacyStoreMethodFinder(importInfo: ParsedImportInfo) {
+  return function filter(value: CallExpression): value is LegacyStoreMethodCallExpression {
+    return (
+      value.callee.type === 'MemberExpression' &&
+      value.callee.property.type === 'Identifier' &&
+      importInfo.importedName === value.callee.property.name &&
+      (value.callee.object.type === 'Identifier' ||
+        (value.callee.object.type === 'MemberExpression' && value.callee.object.property.type === 'Identifier'))
```

Comment:
Look for "store" by default but add config to allow users to pass other names, e.g. `"storeService"` or `"db"`